controllers/nodefeaturediscovery_status.go

```diff
@@ -0,0 +1,611 @@
+			LastHeartbeatTime:  metav1.Time{Time: now},
+		},
+		{
+			Type:               conditionsv1.ConditionUpgradeable,
+			Status:             corev1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+			LastHeartbeatTime:  metav1.Time{Time: now},
+		},
+		{
+			Type:               conditionsv1.ConditionProgressing,
+			Status:             corev1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+			LastHeartbeatTime:  metav1.Time{Time: now},
+		},
+		{
+			Type:               conditionsv1.ConditionDegraded,
+			Status:             corev1.ConditionTrue,
+			LastTransitionTime: metav1.Time{Time: now},
+			LastHeartbeatTime:  metav1.Time{Time: now},
+			Reason:             reason,
+			Message:            message,
+		},
+	}
+}
+
+// getProgressingConditions returns a list of conditionsv1.Condition objects and marks
+// every condition as FALSE except for conditionsv1.ConditionProgressing so that the
+// reconciler can determine that the resource is progressing.
+func (r *NodeFeatureDiscoveryReconciler) getProgressingConditions(reason string, message string) []conditionsv1.Condition {
+	now := time.Now()
+
+	return []conditionsv1.Condition{
+		{
+			Type:               conditionsv1.ConditionAvailable,
+			Status:             corev1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               conditionsv1.ConditionUpgradeable,
+			Status:             corev1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               conditionsv1.ConditionProgressing,
+			Status:             corev1.ConditionTrue,
+			LastTransitionTime: metav1.Time{Time: now},
+			Reason:             reason,
+			Message:            message,
+		},
+		{
+			Type:               conditionsv1.ConditionDegraded,
+			Status:             corev1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+	}
+}
+
+
+// The status of the resource (available, upgradeable, progressing, or
+// degraded).
+type resourceStatus struct {
+
+	// Is the resource available, upgradable, etc.?
+	isAvailable   bool
+	isUpgradeable bool
+	isProgressing bool
+	isDegraded    bool
+
+	// How many statuses are set to 'true'?
+	numActiveStatuses int
+}
+
+// getWorkerDaemonSetConditions is a wrapper around
+// "getDaemonSetConditions" for ease of calling the
+// worker DaemonSet status
+func (r *NodeFeatureDiscoveryReconciler) getWorkerDaemonSetConditions(ctx context.Context) (resourceStatus, error) {
+	return r.__getDaemonSetConditions(ctx, worker)
+}
+
+// getMasterDaemonSetConditions is a wrapper around
+// "getDaemonSetConditions" for ease of calling the
+// master DaemonSet status
+func (r *NodeFeatureDiscoveryReconciler) getMasterDaemonSetConditions(ctx context.Context) (resourceStatus, error) {
+	return r.__getDaemonSetConditions(ctx, master)
+}
+
+func (r *NodeFeatureDiscoveryReconciler) __getDaemonSetConditions(ctx context.Context, node nodeType) (resourceStatus, error) {
```

Comment:
double underscore is not that common in Go.

There are some good resources on this, one I like is https://rakyll.org/style-packages/